### PR TITLE
Dialyzer formatting

### PR DESCRIPTION
**Pull request by asummers**
Improve Dialyzer messaging.

### CODE REVIEW

File: apps/language_server/lib/language_server/dialyzer.ex [hunk 1]
```diff
@@ -442,7 +442,18 @@ defmodule ElixirLS.LanguageServer.Dialyzer do
 
   defp dialyxir_format({_, _, {warning_name, args}}) do
     %{^warning_name => warning_module} = Dialyxir.Warnings.warnings()
-    <<_::binary>> = apply(warning_module, :format_long, [args])
```
Comment (axelson) [inline apps/language_server/lib/language_server/dialyzer.ex]: Is there a reason to drop the `<<_::binary>>` check? I assume it's there to ensure that we're getting a binary back

**Comment by axelson** [comment]
Also we probably want to load dialyxir in the test environment in `apps/language_server/mix.exs` instead of only in `:dev`

`{:dialyxir, "~> 1.0.0-rc.4", only: [:dev, :test], runtime: false}`

File: apps/language_server/lib/language_server/dialyzer.ex (hunk 1 above)
Comment (asummers) [reply apps/language_server/lib/language_server/dialyzer.ex]: The way those functions work, that will always be true. It was simply superfluous. I can put it back though if you feel strongly!

Comment (axelson) [reply apps/language_server/lib/language_server/dialyzer.ex]: I see it more as a guard against it changing in the future, but I guess that's unlikely and I probably wouldn't be arguing for adding it if it wasn't already there ;)

**Pull request merged**